connectors/agent/config.py

```diff
@@ -0,0 +1,105 @@
+from connectors.config import add_defaults
+
+
+class ConnectorsAgentConfigurationWrapper:
+    """A wrapper that facilitates passing configuration from Agent to Connectors Service.
+
+    This class is responsible for:
+    - Storing in-memory configuration of Connectors Service running on Agent
+    - Transforming configuration reported by Agent to valid Connectors Service configuration
+    - Indicating that configuration has changed so that the user of the class can trigger the restart
+    """
+
+    def __init__(self):
+        """Inits the class.
+
+        There's default config that allows us to run connectors natively (see _force_allow_native flag),
+        when final configuration is reported these defaults will be merged with defaults from Connectors
+        Service config and specific config coming from Agent.
+        """
+        self._default_config = {
+            "_force_allow_native": True,
+            "native_service_types": [
+                "azure_blob_storage",
+                "box",
+                "confluence",
+                "dropbox",
+                "github",
+                "gmail",
+                "google_cloud_storage",
+                "google_drive",
+                "jira",
+                "mongodb",
+                "mssql",
+                "mysql",
+                "notion",
+                "onedrive",
+                "oracle",
+                "outlook",
+                "network_drive",
+                "postgresql",
+                "s3",
+                "salesforce",
+                "servicenow",
+                "sharepoint_online",
+                "slack",
+                "microsoft_teams",
+                "zoom",
+            ],
```

Comment:
For starters, should we limit this just to github?

Reply:
This will coming from the agent protocol, for now I guess it's fine to leave this as it is?